src/application/application.cpp

```diff
@@ -226,10 +226,13 @@ void Application::Predict() {
                       config_.precise_float_parser);
     TextReader<int> result_reader(config_.output_result.c_str(), false);
     result_reader.ReadAllLines();
-    std::vector<std::vector<int>> pred_leaf(result_reader.Lines().size());
+
+    std::vector<int> pred_leaf;
+    auto nrow = static_cast<int>(result_reader.Lines().size());
     #pragma omp parallel for num_threads(OMP_NUM_THREADS()) schedule(static)
     for (int i = 0; i < static_cast<int>(result_reader.Lines().size()); ++i) {
-      pred_leaf[i] = Common::StringToArray<int>(result_reader.Lines()[i], '\t');
+      auto line_vec = Common::StringToArray<int>(result_reader.Lines()[i], '\t');
+      pred_leaf.insert(pred_leaf.end(), line_vec.begin(), line_vec.end());
```

Comment:
I am rather confident that this will break when using `#pragma omp parallel`. IMO, there are two possibilities:

- Get rid of the parallelism (which likely results in a considerable performance hit)
- Initialize `pred_leaf` to equal the final size and use index setters. I'm not sure if this is feasible. One idea would be the following: subsequent code assumes that each line in the text file that is being read provides the same number of values. We could read line 0 which yields the number of values per line. Then, we can resize the vector as needed and we can parallelize reading of the remaining lines. If the value count in a line doesn't match the expected count, we can raise an exception.

Reply:
I am pre-allocating the vector now and am using a nested for-loop. Do you still see an issue?

Reply:
No, looks good now :)